rfd/0083-license-expiration-warnings.md

```diff
@@ -0,0 +1,157 @@
+---
+authors: edwarddowling (edward.dowling@goteleport.com), r0mant (roman@goteleport.com)
+---
+
+
+# RFD 83 - License warnings
+
+## Required approvers
+
+- Engineering: `@r0mant`
+- Product: `@klizentas && @xinding33`
+- Security: `@reedloden`
+
+## What
+
+This RFD proposes a way to display license warnings to self hosted users both before and after license expiration.
+
+## Why
+
+To encourage the renewal of licenses after expiration and let users know before their license expires.
+
+## Success criteria
+
+Teleport displays license warnings to users in both CLI and web UI starting from a certain time prior to the license expiration, as well as post-expiration.
+
+## Scope
+
+Self-hosted Teleport will be the target of this RFD. Teleport Cloud is out of scope because the Cloud license does not have proper expiration yet. When Cloud implements proper license expiration, this feature will kick in automatically.
+License warning should be displayed in both Web UI and CLI (tsh/tctl).
+
+### CLI UX
+
+License warnings should be displayed in CLI (tsh/tctl) 
+TSH - on “tsh login”, in “tsh status”
+When the user uses either the “tsh login” or ‘tsh status” commands the appropriate license warning will be displayed. (see examples below)
+TCTL - “tctl status”
+When the user uses the “tctl status” command the appropriate license warning will be displayed.
+The warnings are to be displayed 90 days prior to the license expiring.
+After expiration all tctl commands will display the license expiry warning.
```

Comment:
I was under the impression it was mainly the tsh commands we were trying to avoid adding this extra latency to. Currently i have been working under the assumption that all tctl commands should be checking this. If the extra request adding delays is a big enough concern for tctl as well I could modify it to only check on tctl status similar to how we are handling tsh.

Reply:
I've reworded this section to be clearer